engine/src/main/java/org/destinationsol/game/SolCam.java

```diff
@@ -94,14 +101,14 @@ public void update(SolGame game, float timeStep) {
             }
         }
 
-        if (life < myPrevHeroLife) {
+        if (life < myPrevHeroLife && hull == myPrevHeroHull) {
```

Comment:
Looking just at the code, it is not immediately obvious why this comparison fixes the bug. Could you please add a comment here